### PR TITLE
Show Activity Type for local activities in all SDKs

### PR DESCRIPTION
**Pull request by Alex-Tideman**
## Description & motivation 💭 
When we moved to decoding payloads on demand, we lost the ability to show LocalActivity activityType on the event summary row. This adds back that functionality by decoding the LocalActivity payloads and finding the correct value for ActivityType based on SDK.

### Screenshots (if applicable) 📸 
**AFTER**
Screen Shot 2024-01-22 at 1 42 40 PM

**BEFORE**
Screen Shot 2024-01-12 at 9 25 05 AM

### How was this tested 👻 

- [x] Manual testing
- [ ] E2E tests added
- [x] Unit tests added

### CODE REVIEW

File: src/lib/utilities/get-single-attribute-for-event.ts [hunk 1]
```diff
@@ -208,20 +215,20 @@ const getSummaryAttribute = (event: WorkflowEvent): SummaryAttribute => {
   return first;
 };
 
-export const getSummaryForEventGroup = ({
+export const getSummaryForEventGroup = async ({
   lastEvent,
-}: EventGroup): SummaryAttribute => {
-  return getSummaryAttribute(lastEvent);
+}: EventGroup): Promise<SummaryAttribute> => {
+  return await getSummaryAttribute(lastEvent);
```
Comment (laurakwhit) [inline src/lib/utilities/get-single-attribute-for-event.ts:221]: Is the addition of async/await here needed? `getSummaryAttribute` already returns `Promise<SummaryAttribute>`.

File: src/lib/utilities/get-single-attribute-for-event.ts [hunk 2]
```diff
@@ -208,20 +215,20 @@ const getSummaryAttribute = (event: WorkflowEvent): SummaryAttribute => {
...
-): SummaryAttribute => {
+): Promise<SummaryAttribute> => {
   if (!event) return emptyAttribute;
 
   if (isEventGroup(event)) {
     return getSummaryForEventGroup(event);
   }
 
-  return getSummaryAttribute(event);
+  return await getSummaryAttribute(event);
```
Comment (laurakwhit) [inline src/lib/utilities/get-single-attribute-for-event.ts:233]: Similar comment as above ⬆️

File: src/lib/utilities/get-single-attribute-for-event.ts (hunk 1 above)
Comment (Alex-Tideman) [reply src/lib/utilities/get-single-attribute-for-event.ts]: You are absolutely correct

Comment (laurakwhit) [reply src/lib/utilities/get-single-attribute-for-event.ts]: Think we can remove the async [here](https://github.com/temporalio/ui/pull/1795/commits/910dc77d274b840979e8e393cc19c2f65c54ca2d#diff-42a5d42a3ea1e209202949d938fa2ca7271443c0279d04c2991e2cf844f8e0c5R218) too.

File: src/lib/holocene/select/multi-select.svelte [hunk 3]
```diff
@@ -38,7 +38,12 @@
     } else {
       selectedOptions = [...selectedOptions, option];
     }
-    onChange(selectedOptions);
+
+    if (selectedOptions.length === options.length) {
+      onChange([]);
```
Comment (laurakwhit) [inline src/lib/holocene/select/multi-select.svelte:43]: Wondering if we want this conditional here or in the `onChange`. Might there be an instance were we want all the options to be passed to the `onChange` when they are all selected?

Comment (Alex-Tideman) [reply src/lib/holocene/select/multi-select.svelte:43]: You are absolutely right, I'll move it to the onChange. Good catch

**Review by laurakwhit — APPROVED** [review]

**Pull request merged**